src/main/java/pl/javastart/task/Main.java

```diff
@@ -1,8 +1,23 @@
 package pl.javastart.task;
 
+import java.io.IOException;
+import java.util.List;
+
 public class Main {
 
     public static void main(String[] args) {
+        ProductReader productReader = new ProductReader();
+        List<Product> products = null;
+        try {
+            products = productReader.readAllProducts();
+        } catch (IOException e) {
+            System.err.println("Nie udało się wczytać pliku");
+        }
+        try {
+            productReader.printInfoAboutProductsInEuro(products, new CurrencyReader());
+        } catch (IOException e) {
+            System.err.println("Nie udało się wyświetlić informacji");
+        }
```

Comment:
Zapisałbym to w 1 bloku try catch, jeżeli nie uda się odczytać produktów, to nie chcemy wywoływać metody productReader.printInfoAboutProductsInEuro z nullem.
W code review pokazałem jak można to obsłużyć ładniej wykorzystując np. UncheckedIOException